scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -173,12 +173,15 @@ final class ExplicitResultTypes(
     def hasParentWihTemplate: Boolean =
       defn.parent.exists(_.is[Template])
 
-    isImplicit && !isFinalLiteralVal || {
-      hasParentWihTemplate &&
-      !defn.hasMod(mod"implicit") &&
-      !matchesSimpleDefinition() &&
-      matchesMemberKind() &&
-      matchesMemberVisibility()
+    matchesMemberKind() && matchesMemberVisibility() && {
+      {
+        isImplicit &&
+        !isFinalLiteralVal
+      } || {
+        hasParentWihTemplate &&
+        !defn.hasMod(mod"implicit") &&
+        !matchesSimpleDefinition()
```

Comment:
while you are at it, maybe you can name the remaining these 2 groups? The nesting makes it very hard to read

Reply:
done